dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -266,7 +266,9 @@ class TypedDataset[T](val dataset: Dataset[T])(implicit val encoder: TypedEncode
       encoder: TypedEncoder[Out]
```

Comment:
Please take a look, I've tried to fix a root cause, something similar is done in Spark internally.